### ISSUE TITLE
Add accessors for stats functions

### ISSUE THREAD

**Issue by bradyrx** [bradyrx/esm_analysis#59]
It would be much quicker to have our stats functions registered as accessors as well, as in `xskillscore`. Then one can just do `ds.esm.rm_trend(dim='time')` for instance.

**Comment by luke-gregor** [comment]
Happy to jump in here if this hasn't been taken on yet

**Comment by bradyrx** [comment]
Yes that would be great @luke-gregor! I think I already have the accessor file set up for the `convert_lon` function, so feel free to add the stats functions there. You can follow the model set up in `xskillscore`. I'll look out for a PR. Thanks!

**Comment by luke-gregor** [comment]
I've finally got some time to get to this! Just want to check that anyone else hasn't started on this in the meantime?

**Comment by luke-gregor** [comment]
A question regarding usability. I noticed that for the stats regression functions, that `x` changes from the predicant to the predictor when y is given. Would it not make sense to always have y as the predicant and x as the predictor (if not None)? Seems a little more consistent. 

https://github.com/bradyrx/esmtools/blob/e93391f77f4ffc1bb2cc8844ca7e6f755e283560/esmtools/stats.py#L265-L295

**Comment by aaronspring** [comment]
I wanted to have the current implementation, because I wanted to do `linear_slope(ds, dim='time')`. I'd be happy to keep that way of using it. maybe having the accessor like `ds.esmtools.liner_slope(dim='time')`?

**Comment by luke-gregor** [comment]
Changing this shouldn't change the functionality at all. Just means that `y` will always be the first input argument. Would this make sense? Existing implementations would change though, so might break code for people who have already implemented `linear_slope(x, y=y, dim='time')`

**Comment by aaronspring** [comment]
Ah ok. Please go ahead with a PR. Sounds good to me

**Comment by bradyrx** [comment]
@luke-gregor, my thinking was that in logical order it makes sense to have `x` first, then `y`. I know it's sort of weird to have a singular argument revert to the predictand, since it's technically in the slot order of the predictor otherwise (if I understand your point correctly).

I would say it's intuitively off to have `(y, x=None)` order, although the language is "y regressed *onto* x" so this could make sense. In the end, my defense is that I modeled it after `scipy` conventions: https://docs.scipy.org/doc/scipy/reference/generated/scipy.stats.linregress.html. I could see it going either way, but I see in the PR you got roadblocked by the thorough testing so it seems like we'll keep the order for now?

**Comment by luke-gregor** [comment]
> I could see it going either way, but I see in the PR you got roadblocked by the thorough testing so it seems like we'll keep the order for now?

Yep, exactly. Didn't realise how extensive and impressive the test suite for esmtools is!
